src/formio/base.ts

```diff
@@ -195,3 +195,24 @@ export type InputComponentSchema<
  */
 export interface LayoutComponentSchema<T = never>
   extends Omit<ComponentSchema<T>, UnusedFormioProperties> {}
+
+/**
+ * @group Schema primitives
+ */
+export interface TableViewComponentSchema<T = Record<string, any>>
```

Comment:
```suggestion
export interface TableViewComponentSchema<T extends JSONObject = JSONObject>
```
I'd say that we expect every row of the table to be an object, not just any record. The values are all JSONSerializable, otherwise we cannot send them over the wire through the API, so `any` is definitely not right here.

And, if you really don't know what the type upfront is, it's usually better to use `unknown` instead of `any`.